### PR TITLE
Add default configuration files

### PR DESCRIPTION
**Pull request by HashidaTKS**
The default configuration files are Common.txt, TrustedDomains.txt, UnsafeDomains.txt and UnsafeFiles.txt in `{app}\DefaultConfig`. Those files are installed when they are located in the "DefaultConfig" folder that exists in the same folder to installer like below.

```
│  FlexConfirmMailSetup-22.6.0.exe
│
└─DefaultConfig
        Common.txt
        TrustedDomains.txt
        UnsafeDomains.txt
```

The configurations for trusted domains, unsafe domains and unsafe files in the policy configuration and the default configuration files are fixed, they can't modified by user configuration files. The priority of common configurations is as following (left is high).

User file configurations > Default file configurations > Policy configurations

Note that configurations by default file is regarded as specified by organization.

Here are sample images.
In these images, `from_registry_*` are configurations from registry,  `default_config_*` are configrations from default files and others are configurations from user configs.

image

image(1)

image

## Test

Prerequisites:

* Default Config:  `C:\Program Files\FlexConfirmMail\DefaultConfig\*.txt`
* User Config: `%UserAppData%\FlexConfirmMail\*.txt`
* Registry Config: Config specified by GPO
* "Merge" means:
  * For Trusted Domains, Unsafe Domains, Unsafe Files
    * All configurations of the policy configuration, the default configuration and the user configuration are used.
    *  The configurations in the policy configuration and the default configuration files are fixed, they can't modified by user 
  * For Common Config
    *  If a setting is modified from default value, the highest priority config are used. The priority is as following.
    * User file configurations > Default file configurations > Policy configurations.

Test:

* [x] Confirm that `DefaultConfig/*.txt` is installed to `C:\Program Files\FlexConfirmMail\DefaultConfig\*.txt`.
* [x] Confirm that default config is used if there is only default config (remove other settings if exists).
* [x] Confirm that registry config is used if there is only registry config  (remove other settings if exists).
* [x] Confirm that user config is used if there is only user config  (remove other settings if exists).
* [x] Confirm that registry config and default config are merged if they are specified.
* [x] Confirm that default config and user config are merged if they are specified.
* [x] Confirm that registry config and user config are merged if they are specified.
* [x] Confirm that all configs are merged if they are specified.

### CODE REVIEW

File: FlexConfirmMail.iss [hunk 1]
```diff
@@ -42,8 +42,9 @@ Source: "bin\Release\FlexConfirmMail.dll"; DestDir: "{app}"; Flags: ignoreversio
 Source: "bin\Release\FlexConfirmMail.dll.manifest"; DestDir: "{app}"; Flags: ignoreversion
 Source: "bin\Release\FlexConfirmMail.vsto"; DestDir: "{app}"; Flags: ignoreversion
 Source: "bin\Release\en\FlexConfirmMail.resources.dll"; DestDir: "{app}\en"; Flags: ignoreversion
-Source: "bin\Release\zh\FlexConfirmMail.resources.dll"; DestDir: "{app}\zh"; Flags: ignoreversion
+Source: "bin\Release\zh\FlexConfirmMail.resources.dll"; DestDir: "{app}\zh"; Flags: ignoreversion
```
Comment (HashidaTKS) [inline FlexConfirmMail.iss:45]: 改行コードを統一したことによる差分と思われる。

File: Dialog/ConfigDialog.xaml.cs [hunk 2]
```diff
@@ -24,7 +24,10 @@ public ConfigDialog()
             {
                 _default = Loader.LoadFromReg(RegistryPath.DefaultPolicy);
                 _config.Merge(_default);
-            }
+            }
+            Config defaultWithFile = Loader.LoadFromDir(StandardPath.GetDefaultConfigDir());
+            _default.Merge(defaultWithFile);
```
Comment (HashidaTKS) [inline Dialog/ConfigDialog.xaml.cs:29]: DefaultConfigによる設定はレジストリーによる設定と同様、組織による設定の指定とみなしている。

Comment (piroor) [reply Dialog/ConfigDialog.xaml.cs:29]: PRの説明が、ポリシー設定が最も優先度が低いと書かれているように読めましたが、ここは確かにその説明に則っていると思えました。
また、次の行で_configに対してdefaultWithFileをMergeしていますが、こちらも、ポリシーの内容を適用→defaultWithFileの内容を適用→_localの内容を適用 という順ということで、PRの説明に合致していると理解しました。

**Review by piroor — APPROVED** [review]
自分が見た限り、PRの説明との間に齟齬はないように感じました。

**Comment by HashidaTKS** [comment]
確認ありがとうございます！

**Pull request merged**